### DISCUSSION TITLE
How does sqlalchemy connect to the cluster

### DISCUSSION THREAD

**Discussion by phpsxg** [sqlalchemy/sqlalchemy#7528] [Usage Questions] [answered]
The code for sqlalchemy to connect to a single machine is as follows：
`conn_str = 'clickhouse://default:@localhost/default'
engine = create_engine(conn_str)`

How does sqlalchemy connect to the cluster?Thanks~

**Comment by CaselIT** [comment, marked as answer]
Hi,

SQLAlchemy does not support clickhouse directly so I think you should ask to the clickhouse sqlalchemy plugin developers.
